core/models.py

```diff
@@ -0,0 +1,35 @@
+from django.db import models
+from django.contrib.auth.models import AbstractUser
+from django.db.models.constraints import UniqueConstraint
+
+from datetime import date
+
+class User(AbstractUser):
+    def __repr__(self):
+        return f"<User username={self.username}>"
+
+    def __str__(self):
+        return self.username
+
+
+class Habit(models.Model):
+    user = models.ForeignKey(to=User, on_delete=models.CASCADE)
+    name = models.CharField(max_length=256)
+    target = models.PositiveIntegerField()
+    created_date = models.DateField(default=date.today)
+
+    def __repr__(self):
+        return f"<Habit title={self.title}>"
+
+    def __str__(self):
+        return self.title
+
+class Record(models.Model):
+    habit = models.ForeignKey(Habit, null=True, on_delete=models.CASCADE, related_name="records")
+    outcome = models.PositiveIntegerField()
+    date = models.DateField()
+
+    class Meta:
+        constraints = [
+            UniqueConstraint(fields=['habit', 'outcome', 'date'], name='unique_records')
```

Comment:
You don't need three fields here; it is enough to have a unique record per date for the requirements of the assignment.